libs/barista-components/breadcrumbs/src/breadcrumbs.ts

```diff
@@ -49,38 +98,394 @@ export const _DtBreadcrumbMixinBase = mixinColor<
+        }
+
+        if (toTransplant.length > 0 || toPutBack.length > 0) {
+          // If items are transplanted or put back, mark the breadcrumbs dirty
+          this._changeDetectorRef.markForCheck();
+        }
+      });
+
+    if ('ResizeObserver' in window) {
+      this._containerSizeObserver = new window.ResizeObserver((entries) => {
+        if (entries && entries[0]) {
+          // The next call needs to be run inside the zone otherwise
+          // all code that is executed due to the emition would also run
+          // outside zone. The resizeobserver is not something zone.js is listening
+          // to.
+          this._zone.run(() => {
+            this._containerContentRect$.next(entries[0].contentRect);
+          });
+        }
+      });
+
+      this._containerSizeObserver.observe(this._elementRef.nativeElement);
+    }
   }
 
   ngAfterContentInit(): void {
     this._items.changes
       .pipe(startWith(null), takeUntil(this._destroy$))
       .subscribe(() => {
-        // We need to notify the items whether they are the last one in the list,
-        // because they use this information to determine their active state.
+        // If items change, reset everything to default
+        this._itemsWidthMap.clear();
+
+        if (this._itemsPortalsMap.size > 0) {
+          // Move all transplanted items back from the overlay to the breadcrumbs
+          // so that each item's width can be calculated correctly again
+          this._itemsPortalsMap.forEach((item) => {
+            item.detach();
+          });
+
+          this._itemsPortalsMap.clear();
+          this._overlayRef?.detach();
+          // Mark breadcrumbs dirty to correctly render the overlay toggle
+          this._changeDetectorRef.markForCheck();
+        }
+
         this._items.forEach((item, index) => {
+          item._setEllipsis(false);
+          // We need to notify the items whether they are the last one in the list,
+          // because they use this information to determine their active state.
           item._setCurrent(this._items.length - 1 === index);
+          this._itemsWidthMap.set(
+            item,
+            item._elementRef.nativeElement.getBoundingClientRect().width,
+          );
         });
       });
   }
 
   ngOnDestroy(): void {
     this._destroy$.next();
     this._destroy$.complete();
+    if (this._overlayRef) {
+      this._overlayRef.dispose();
+    }
+  }
+
+  closeOverlay(): void {
+    this._overlayRef?.detach();
+  }
+
+  _createOverlay(): void {
+    if (!this._overlayRef) {
+      const overlayConfig: OverlayConfig = {
+        positionStrategy: this._overlay
+          .position()
+          .flexibleConnectedTo(this._elementRef)
+          .setOrigin(this._elementRef)
+          .withPositions(DT_BREADCRUMBS_OVERLAY_POSITIONS),
+        panelClass: 'dt-breadcrumbs-overlay',
+      };
+      this._overlayRef = this._overlay.create(overlayConfig);
+
+      this._overlayRef
+        ?.keydownEvents()
```

Comment:
Just a note - I've checked the stream completes when the overlay is disposed. So we are safe here regarding unsubscribing.